Source/WebCore/style/StyleBuilderState.h

```diff
@@ -83,9 +83,8 @@ class BuilderState {
     inline const FontCascadeDescription& fontDescription();
     inline const FontCascadeDescription& parentFontDescription();
 
-    // FIXME: These are mutually exclusive, clean up the code to take that into account.
     bool applyPropertyToRegularStyle() const { return m_linkMatch != SelectorChecker::MatchVisited; }
-    bool applyPropertyToVisitedLinkStyle() const { return m_linkMatch == SelectorChecker::MatchVisited; }
+    bool applyPropertyToVisitedLinkStyle() const { return m_linkMatch != SelectorChecker::MatchLink; }
```

Comment:
We should also (separately) fix all the applyInheritFoo functions with visited link handling (generated and otherwise) to inherit from the visited link style.